findspam.py

```diff
@@ -316,6 +321,20 @@ def bad_pattern_in_url(s, site, *args):
         return False, ""
 
 
+def bad_ns_for_url_domain(s, site, *args):
+    for domain in set([get_domain(link) for link in post_links(s)]):
+        try:
+            ns = dns.resolver.query(domain, 'ns')
+        except dns.exception.DNSException as exc:
+            # TODO: log exception?
+            continue
+        nameservers = [server.target.to_text() for server in ns]
+        if any([ns.endswith('.namecheaphosting.com.') for ns in nameservers]):
```

Comment:
Agree with @tripleee on this one. Do the simplest thing that might work first; from here we have two easy paths: (1) rollback, or (2) add a rule field for it. Seems like a pretty good idea to me.